include/dgl/runtime/ndarray.h

```diff
@@ -292,16 +292,18 @@ struct NDArray::Container {
 // the usages of functions are documented in place.
 inline NDArray::NDArray(Container* data)
   : data_(data) {
-  data_->IncRef();
+  if (data_)
+    data_->IncRef();
```

Comment:
do you see a case that `data_` is nullptr?

Reply:
Yes, see following codes:
```c++
NDArray arr;  // This NDArray contains null data_
NDArray arr2 = arr;  // This will call the copy constructor above and segfault if not correctly protected.
```